crates/matrix-sdk/src/oidc/cross_process.rs

```diff
@@ -7,34 +7,28 @@ use matrix_sdk_base::crypto::{
 use matrix_sdk_common::store_locks::{
     CrossProcessStoreLock, CrossProcessStoreLockGuard, LockStoreError,
 };
+use sha2::{Digest as _, Sha256};
 use thiserror::Error;
 use tokio::sync::{Mutex, OwnedMutexGuard};
 use tracing::trace;
 
 use super::OidcSessionTokens;
-use crate::oidc::hash;
 
 /// Key in the database for the custom value holding the current session tokens
 /// hash.
 const OIDC_SESSION_HASH_KEY: &str = "oidc_session_hash";
 
 /// Newtype to identify that a value is a session tokens' hash.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-struct SessionHash(u64);
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct SessionHash(Vec<u8>);
 
 /// Compute a hash uniquely identifying the OIDC session tokens.
 fn compute_session_hash(tokens: &OidcSessionTokens) -> SessionHash {
-    // Subset of `SessionTokens` fit for hashing
-    #[derive(Hash)]
-    struct HashableSessionTokens<'a> {
-        access_token: &'a str,
-        refresh_token: Option<&'a str>,
+    let mut hash = Sha256::new().chain_update(tokens.access_token.as_bytes());
+    if let Some(refresh_token) = &tokens.refresh_token {
+        hash = hash.chain_update(refresh_token.as_bytes());
     }
-
-    SessionHash(hash(&HashableSessionTokens {
-        access_token: &tokens.access_token,
-        refresh_token: tokens.refresh_token.as_deref(),
-    }))
+    SessionHash(hash.finalize().as_slice().to_owned())
```

Comment:
Does [`into_vec()`](https://docs.rs/generic-array/latest/generic_array/struct.GenericArray.html#method.into_vec) not work here?

Reply:
Ah yes, `finalize()` returns a `GenericArray` which has a `Deref<Output = &[]>`, so I can use `to_vec()` directly :+1: